internal/storagenode/logstream/subscribe.go

```diff
@@ -121,6 +128,12 @@ func (lse *Executor) SubscribeWithLLSN(begin, end types.LLSN) (*SubscribeResult,
 	if begin < localLowWatermark.LLSN {
 		return nil, fmt.Errorf("log stream: %w", verrors.ErrTrimmed)
 	}
+	_, _, uncommittedBegin, _ := lse.lsc.reportCommitBase()
+	if localLowWatermark.LLSN.Invalid() && end <= uncommittedBegin.LLSN {
+		return nil, fmt.Errorf("log stream: %w", verrors.ErrTrimmed)
+		//sr, _ := lse.newEmptySubscribeResult()
+		//return sr, nil
```

Comment:
Why is return of SubscribeWithLLSN different from the return of SubscribeWithGLSN?

Reply:
You already mentioned.

`In addition, the log stream can't know the global low watermark after a restart because it keeps it
in memory. Therefore, Subscribe API can't decide whether it returns an empty result or error
indicating that the log entries are already trimmed, and it needs some discussion for a design
decision.`